kubernetes-client/src/test/java/io/fabric8/kubernetes/client/dsl/internal/RawCustomResourceOperationsImplTest.java

```diff
@@ -75,15 +76,42 @@ void testCreateOrReplaceUrl() throws IOException {
     String resourceAsString = "{\"metadata\":{\"name\":\"myresource\",\"namespace\":\"myns\"}, \"kind\":\"raw\", \"apiVersion\":\"v1\"}";
     ArgumentCaptor<Request> captor = ArgumentCaptor.forClass(Request.class);
 
+    Call mockCall = mock(Call.class);
+    Response mockErrorResponse = mockResponse(500);
+    Response mockConflictResponse = mockResponse(409);
+    when(mockCall.execute())
+      .thenReturn(mockErrorResponse, mockConflictResponse, mockSuccessResponse);
+    when(mockClient.newCall(any())).thenReturn(mockCall);
+
     // When
+    try {
+      rawCustomResourceOperations.createOrReplace(resourceAsString);
+      fail("expected first call to createOrReplace to throw exception due to 500 response");
+    } catch (KubernetesClientException e) {
+      assertEquals(500, e.getCode());
+    }
     rawCustomResourceOperations.createOrReplace(resourceAsString);
     rawCustomResourceOperations.createOrReplace("myns", resourceAsString);
 
     // Then
-    verify(mockClient, times(2)).newCall(captor.capture());
-    assertEquals(2, captor.getAllValues().size());
+    verify(mockClient, times(4)).newCall(captor.capture());
+    assertEquals(4, captor.getAllValues().size());
     assertEquals("/apis/test.fabric8.io/v1alpha1/hellos", captor.getAllValues().get(0).url().encodedPath());
-    assertEquals("/apis/test.fabric8.io/v1alpha1/namespaces/myns/hellos", captor.getAllValues().get(1).url().encodedPath());
+    assertEquals("POST", captor.getAllValues().get(0).method());
+    assertEquals("/apis/test.fabric8.io/v1alpha1/hellos", captor.getAllValues().get(1).url().encodedPath());
+    assertEquals("POST", captor.getAllValues().get(1).method());
+    assertEquals("/apis/test.fabric8.io/v1alpha1/hellos/myresource", captor.getAllValues().get(2).url().encodedPath());
+    assertEquals("PUT", captor.getAllValues().get(2).method());
+    assertEquals("/apis/test.fabric8.io/v1alpha1/namespaces/myns/hellos", captor.getAllValues().get(3).url().encodedPath());
+    assertEquals("POST", captor.getAllValues().get(3).method());
+  }
+
+  private Response mockResponse(int code) {
+    Response mockResponse = mock(Response.class);
+    when(mockResponse.isSuccessful()).thenReturn(code < 400);
```

Comment:
nit, maybe use `HttpUrlConnection` for status codes:
```suggestion
    when(mockResponse.isSuccessful()).thenReturn(code < HttpURLConnection.HTTP_BAD_REQUEST);
```

Reply:
Instead I updated to just build a Response object directly so we can re-use the internal isSuccessful method which does this comparison for us. I updated the other codes to use HttpURLConnection though.